src/nvapi_sys.cpp

```diff
@@ -89,4 +89,48 @@ extern "C" {
 
         return Ok(n);
     }
+
+    NvAPI_Status __cdecl NvAPI_SYS_GetPhysicalGPUs(NV_PHYSICAL_GPUS* pPhysicalGPUs) {
+        constexpr auto n = __func__;
+
+        if (log::tracing())
+            log::trace(n, log::fmt::ptr(pPhysicalGPUs));
+
+        if (nvapiAdapterRegistry == nullptr)
+            return ApiNotInitialized(n);
+
+        if (pPhysicalGPUs == nullptr)
+            return InvalidArgument(n);
+
+        for (auto i = 0U; i < nvapiAdapterRegistry->GetAdapterCount(); i++) {
```

Comment:
We should check the struct version beforehand. Would be a good idea to verify Windows behavior here.

Same in `NvAPI_SYS_GetLogicalGPUs`.

Reply:
Thanks, forget about the versioning, I've added the usual ceremony.